daemon/system-info.sh

```diff
@@ -462,7 +462,7 @@ if [ "${VIRTUALIZATION}" != "none" ] && command -v curl > /dev/null 2>&1; then
 
     # Try GCE computeMetadata v1
     if [ "${CLOUD_TYPE}" = "unknown" ]; then
-      if [ -n "$(curl --fail -s --connect-timeout 1 -m 3 --noproxy "*" -H "Metadata-Flavor: Google" "http://metadata.google.internal/computeMetadata/v1")" ]; then
+      if data="$(curl --fail -s --connect-timeout 1 -m 3 --noproxy "*" -H "Metadata-Flavor: Google" "http://metadata.google.internal/computeMetadata/v1")" && echo "$data" | grep -sq computeMetadata; then
```

Comment:
```suggestion
      if curl --fail -s --connect-timeout 1 -m 3 --noproxy "*" -H "Metadata-Flavor: Google" "http://metadata.google.internal/computeMetadata/v1" grep -sq computeMetadata; then
```

If we’re not actually reusing the output, it makes more sense to just pipe it directly to `grep`. This will also ensure that we fail a bit more cleanly if the `curl` command does not succeed.

Reply:
It is already clear. If `curl` fails **or** its exit code is not 0 (HTTP resp code != 200) we don't do echo/grep.

Reply:
Not a big deal. You can "commit suggestion" if you want, but please red button the PR after.